Tests/Opc.Ua.Client.Tests/ContinuationPointInBatchTest.cs

```diff
@@ -0,0 +1,1389 @@
+            theSession.UseManagedBrowseInFetchReferences = UseManagedBrowse;
+
+            // the ExpectedNumber* parameters are not relevant/correct for this test.
+            ManagedBrowseExpectedResultValues pass1ExpectedResults = new ManagedBrowseExpectedResultValues {
+                InputMaxNumberOfContinuationPoints = testData.MaxNumberOfContinuationPoints,
+                InputMaxNumberOfReferencesPerNode = testData.MaxNumberOfReferencesPerNode,
+                ExpectedNumberOfPasses = testData.ExpectedNumberOfPasses,
+                ExpectedNumberOfBadNoCPSCs = testData.ExpectedNumberOfBadNoCPSCs
+            };
+
+            var result = new List<INode>();
+            var nodesToBrowse = new ExpandedNodeIdCollection {
+                ObjectIds.ObjectsFolder
+            };
+
+            Session.FetchTypeTree(ReferenceTypeIds.References);
+            var referenceTypeIds = new NodeIdCollection() { ReferenceTypeIds.HierarchicalReferences };
+            while (nodesToBrowse.Count > 0)
+            {
+                var nextNodesToBrowse = new ExpandedNodeIdCollection();
+                try
+                {
+                    var organizers = Session.NodeCache.FindReferences(
+                        nodesToBrowse,
+                        referenceTypeIds,
+                        false,
+                        true);
+                    nextNodesToBrowse.AddRange(organizers.Select(n => n.NodeId));
+                    var objectNodes = organizers.Where(n => n is ObjectNode);
+                    var variableNodes = organizers.Where(n => n is VariableNode);
+                    result.AddRange(variableNodes);
+                }
+                catch (ServiceResultException sre)
+                {
+                    if (sre.StatusCode == StatusCodes.BadUserAccessDenied)
+                    {
+                        TestContext.Out.WriteLine($"Access denied: Skipped node.");
+                    }
+                }
+                nodesToBrowse = new ExpandedNodeIdCollection(nextNodesToBrowse.Distinct());
+                TestContext.Out.WriteLine("Found {0} duplicates", nextNodesToBrowse.Count - nodesToBrowse.Count);
+            }
+
+            TestContext.Out.WriteLine("Found {0} variables", result.Count);
+        }
+
+        /// <summary>
+        /// Test concurrent access of FetchNodes.
+        /// This test fails. The test is running the browse/browse next itself, and creates
+        /// too many continuation points (the server responds with BadTooManyOperations).
+        /// </summary>
+        [Theory]
+        [Test, Order(1000)]
+        [Ignore("This test case ignores the server setting for the maximal number of " +
+            "Continuation Points/nodes per browse request in this setup and will fail " +
+            "with BadTooManyOperations. It's here to demonstrate this fact.")]
+        public void NodeCacheFetchNodesConcurrent(bool UseManagedBrowse)
```

Comment:
why not make this test pass with Assert.Throws<ServiceResultException> with statuscode BadToManyOperations?

Reply:
Because I'm not sure whether this behaviour can be reproduced in each run with any CPU, and because I want to discuss whether this method should be kept.

Reply:
I removed this testcase.